[BUG] GD32 MFL cannot build with BINARY_FILE_TRANSFER

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

Marlin GD32 MFL doesn't compile with my configuration files.

### Bug Timeline

_No response_

### Expected behavior

Successful build

### Actual behavior

Errors with build in various files.

### Steps to Reproduce

_No response_

### Version of Marlin Firmware

bugfix-2.1.x

### Printer model

Ender-3

### Electronics

GD32

### LCD/Controller

Stock

### Other add-ons

_No response_

### Bed Leveling

UBL Bilinear mesh

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [x] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[CrealityV422MFL-PRO.zip](https://github.com/user-attachments/files/19774197/CrealityV422MFL-PRO.zip)

This is a log when compiling with modified Configuration.h , Configuration_adv.h
With base files it compiles without problems
...............
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\bedlevel\G26.cpp.o
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\bedlevel\G42.cpp.o
Marlin\src\feature\babystep.cpp: In static member function 'static void Babystep::step_axis(AxisEnum)':
Marlin\src\feature\babystep.cpp:53:45: warning: '--' expression of 'volatile'-qualified type is deprecated [-Wvolatile]
53 | if (curTodo > 0) steps[BS_AXIS_IND(axis)]--; else steps[BS_AXIS_IND(axis)]++;
| ~~~~~~~~~~~~~~~~~~~~~~~^
Marlin\src\feature\babystep.cpp:53:78: warning: '++' expression of 'volatile'-qualified type is deprecated [-Wvolatile]
53 | if (curTodo > 0) steps[BS_AXIS_IND(axis)]--; else steps[BS_AXIS_IND(axis)]++;
| ~~~~~~~~~~~~~~~~~~~~~~~^
In file included from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\cores\arduino/Arduino.h:7,
from Marlin\src\feature../sd/../inc/../HAL/../HAL/GD32_MFL/../shared/Marduino.h:36,
from Marlin\src\feature../sd/../inc/../HAL/../HAL/GD32_MFL/HAL.h:27,
from Marlin\src\feature../sd/../inc/../HAL/HAL.h:30,
from Marlin\src\feature../sd/../inc/MarlinConfigPre-5-post.h:29,
from Marlin\src\feature../sd/../inc/MarlinConfigPre-6-type.h:24,
from Marlin\src\feature../sd/../inc/MarlinConfig.h:28,
from Marlin\src\feature../sd/cardreader.h:28,
from Marlin\src\feature\binary_stream.cpp:27:
C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\cores\arduino/CoreHandler.h:10:25: error: expected unqualified-id before numeric constant 10 | #define VERSION_MAJOR 1 // Major version
| ^
Marlin\src\feature\binary_stream.h:197:25: note: in expansion of macro 'VERSION_MAJOR'
197 | static const uint16_t VERSION_MAJOR = 0, VERSION_MINOR = 1, VERSION_PATCH = 0, TIMEOUT = 10000, IDLE_PERIOD = 1000;
| ^~~~~~~~~~~~~
In file included from Marlin\src\feature\binary_stream.cpp:28:
Marlin\src\feature\binary_stream.h: In static member function 'static void SDFileTransferProtocol::idle()':
Marlin\src\feature\binary_stream.h:140:27: error: 'IDLE_PERIOD' was not declared in this scope
140 | idle_timeout = ms + IDLE_PERIOD;
| ^~~~~~~~~~~
Marlin\src\feature\binary_stream.h: In static member function 'static void SDFileTransferProtocol::process(uint8_t, char*, uint16_t)':
Marlin\src\feature\binary_stream.h:146:35: error: 'TIMEOUT' was not declared in this scope
146 | transfer_timeout = millis() + TIMEOUT;
| ^~~~~~~
Marlin\src\feature\binary_stream.h:146:35: note: suggested alternatives:
In file included from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\system\MFL\MFL\Source\FMC/FMC.hpp:24,
from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\cores\arduino/mfl_api.h:47,
from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\cores\arduino/Arduino.h:10:
C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\system\MFL\MFL\Source\FMC/fmc_config.hpp:280:5: note: 'fmc::FMC_Error_Type::TIMEOUT'
280 | TIMEOUT
| ^~~~~~~
In file included from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\system\MFL\MFL\Source\I2C/I2C.hpp:25,
from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\cores\arduino/mfl_api.h:53:
C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\system\MFL\MFL\Source\I2C/i2c_config.hpp:318:5: note: 'i2c::I2C_Error_Type::TIMEOUT'
318 | TIMEOUT,
| ^~~~~~~
In file included from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\system\MFL\MFL\Source\USART/USART.hpp:25,
from C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\cores\arduino/mfl_api.h:68:
C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\system\MFL\MFL\Source\USART/usart_config.hpp:315:5: note: 'usart::USART_Error_Type::TIMEOUT'
| ^~~~~~~
Marlin\src\feature\binary_stream.h: At global scope:
C:\Users\Lazarov.platformio\packages\framework-arduino-mfl\cores\arduino/CoreHandler.h:10:25: error: expected unqualified-id before numeric constant 10 | #define VERSION_MAJOR 1 // Major version
| ^
Marlin\src\feature\binary_stream.h:449:84: note: in expansion of macro 'VERSION_MAJOR'
449 | static const uint16_t PACKET_MAX_WAIT = 500, RX_TIMESLICE = 20, MAX_RETRIES = 0, VERSION_MAJOR = 0, VERSION_MINOR = 1, VERSION_PATCH = 0;
| ^~~~~~~~~~~~~
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\bedlevel\M420.cpp.o
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\bedlevel\ubl\G29.cpp.o
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\bedlevel\ubl\M421.cpp.o
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\calibrate\G28.cpp.o
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\config\M200-M205.cpp.o
Compiling .pio\build\GD32F303RE_creality_mfl\src\src\gcode\config\M220.cpp.o
*** [.pio\build\GD32F303RE_creality_mfl\src\src\feature\binary_stream.cpp.o] Error 1
============================================================ [FAILED] Took 21.58 seconds ============================================================

Environment Status Duration

GD32F303RE_creality_mfl FAILED 00:00:21.577
======================================================= 1 failed, 0 succeeded in 00:00:21.577 =======================================================

--- a/Marlin/src/feature/binary_stream.h
+++ b/Marlin/src/feature/binary_stream.h
@@ -134,7 +134,7 @@ private:
 public:
 
   static void idle() {
-    // If a transfer is interrupted and a file is left open, abort it after timeout ms
+    // If a transfer is interrupted and a file is left open, abort it after 'idle_period' ms
     const millis_t ms = millis();
     if (transfer_active && ELAPSED(ms, idle_timeout)) {
       idle_timeout = ms + idle_period;

--- a/Marlin/src/feature/binary_stream.h
+++ b/Marlin/src/feature/binary_stream.h
@@ -134,19 +134,19 @@ private:
 public:
 
   static void idle() {
-    // If a transfer is interrupted and a file is left open, abort it after TIMEOUT ms
+    // If a transfer is interrupted and a file is left open, abort it after timeout ms
     const millis_t ms = millis();
     if (transfer_active && ELAPSED(ms, idle_timeout)) {
-      idle_timeout = ms + IDLE_PERIOD;
+      idle_timeout = ms + idle_period;
       if (ELAPSED(ms, transfer_timeout)) transfer_abort();
     }
   }
 
   static void process(uint8_t packet_type, char *buffer, const uint16_t length) {
-    transfer_timeout = millis() + TIMEOUT;
+    transfer_timeout = millis() + timeout;
     switch (static_cast<FileTransfer>(packet_type)) {
       case FileTransfer::QUERY:
-        SERIAL_ECHO(F("PFT:version:"), VERSION_MAJOR, C('.'), VERSION_MINOR, C('.'), VERSION_PATCH);
+        SERIAL_ECHO(F("PFT:version:"), version_major, C('.'), version_minor, C('.'), version_patch);
         #if ENABLED(BINARY_STREAM_COMPRESSION)
           SERIAL_ECHOLN(F(":compression:heatshrink,"), HEATSHRINK_STATIC_WINDOW_BITS, C(','), HEATSHRINK_STATIC_LOOKAHEAD_BITS);
         #else
@@ -194,7 +194,7 @@ public:
     }
   }
 
-  static const uint16_t VERSION_MAJOR = 0, VERSION_MINOR = 1, VERSION_PATCH = 0, TIMEOUT = 10000, IDLE_PERIOD = 1000;
+  static const uint16_t version_major = 0, version_minor = 1, version_patch = 0, timeout = 10000, idle_period = 1000;
 };
 
 class BinaryStream {
@@ -209,7 +209,7 @@ public:
   struct Packet { // 10 byte protocol overhead, ascii with checksum and line number has a minimum of 7 increasing with line
 
     union Header {
-      static constexpr uint16_t HEADER_TOKEN = 0xB5AD;
+      static constexpr uint16_t header_token = 0xB5AD;
       struct [[gnu::packed]] {
         uint16_t token;       // packet start token
         uint8_t sync;         // stream sync, resend id and packet loss detection
@@ -245,7 +245,7 @@ public:
       bytes_received = 0;
       checksum = 0;
       header_checksum = 0;
-      timeout = millis() + PACKET_MAX_WAIT;
+      timeout = millis() + packet_max_wait;
       buffer = nullptr;
     }
   } packet{};
@@ -272,14 +272,14 @@ public:
     }
     if (!bs_serial_data_available(card.transfer_port_index)) return false;
     data = bs_read_serial(card.transfer_port_index);
-    packet.timeout = millis() + PACKET_MAX_WAIT;
+    packet.timeout = millis() + packet_max_wait;
     return true;
   }
 
   template<const size_t buffer_size>
   void receive(char (&buffer)[buffer_size]) {
     uint8_t data = 0;
-    millis_t transfer_window = millis() + RX_TIMESLICE;
+    millis_t transfer_window = millis() + rx_timeslice;
 
     #if HAS_MEDIA
       PORT_REDIRECT(SERIAL_PORTMASK(card.transfer_port_index));
@@ -299,7 +299,7 @@ public:
         case StreamState::PACKET_WAIT:
           if (!stream_read(data)) { idle(); return; }  // no active packet so don't wait
           packet.header.data[1] = data;
-          if (packet.header.token == packet.header.HEADER_TOKEN) {
+          if (packet.header.token == packet.header.header_token) {
             packet.bytes_received = 2;
             stream_state = StreamState::PACKET_HEADER;
           }
@@ -322,7 +322,7 @@ public:
             if (packet.header.checksum == packet.header_checksum) {
               // The SYNC control packet is a special case in that it doesn't require the stream sync to be correct
               if (static_cast<Protocol>(packet.header.protocol()) == Protocol::CONTROL && static_cast<ProtocolControl>(packet.header.type()) == ProtocolControl::SYNC) {
-                  SERIAL_ECHOLN(F("ss"), sync, C(','), buffer_size, C(','), VERSION_MAJOR, C('.'), VERSION_MINOR, C('.'), VERSION_PATCH);
+                  SERIAL_ECHOLN(F("ss"), sync, C(','), buffer_size, C(','), version_major, C('.'), version_minor, C('.'), version_patch);
                   stream_state = StreamState::PACKET_RESET;
                   break;
               }
@@ -398,7 +398,7 @@ public:
           stream_state = StreamState::PACKET_RESET;
           break;
         case StreamState::PACKET_RESEND:
-          if (packet_retries < MAX_RETRIES || MAX_RETRIES == 0) {
+          if (packet_retries < max_retries || max_retries == 0) {
             packet_retries++;
             stream_state = StreamState::PACKET_RESET;
             SERIAL_ECHO_MSG("Resend request ", packet_retries);
@@ -446,7 +446,7 @@ public:
     SDFileTransferProtocol::idle();
   }
 
-  static const uint16_t PACKET_MAX_WAIT = 500, RX_TIMESLICE = 20, MAX_RETRIES = 0, VERSION_MAJOR = 0, VERSION_MINOR = 1, VERSION_PATCH = 0;
+  static const uint16_t packet_max_wait = 500, rx_timeslice = 20, max_retries = 0, version_major = 0, version_minor = 1, version_patch = 0;
   uint8_t  packet_retries, sync;
   uint16_t buffer_next_index;
   uint32_t bytes_received;

